exec/java-exec/src/main/java/org/apache/drill/exec/record/AbstractSingleRecordBatch.java

```diff
@@ -61,6 +63,10 @@ public IterOutcome innerNext() {
     }
     switch (upstream) {
     case NONE:
+      if (state == BatchState.FIRST) {
+        return handleFastNone();
```

Comment:
Again, consider a more descriptive name, such as `handleEmptyInput()`. Otherwise, everyone has to remember that "fast none" == "empty input".

Reply:
Changed to `handleNullInput`